apps/site/src/components/pages/hub/block-data-container.tsx

```diff
@@ -53,143 +70,218 @@ export const BlockDataContainer: FunctionComponent<BlockDataContainerProps> = ({
-      const example = {
-        ...metadata?.examples?.[0],
-        ...metadata?.variants?.[0]?.examples?.[0],
-        ...metadata?.variants?.[0]?.properties,
+    const interval = intervalForAllTime();
+
+    // Without a schema, or a full entity (provided within the `exampleGraph`) we're unable to infer what a correct
+    // entity looks like, and as such we can't continue
+    if (metadata.schema) {
+      // Create the boilerplate of an entity
+      const exampleEntity: Entity | EntityTemporal = {
+        properties: {},
+        metadata: {
+          recordId: {
+            entityId: `test-entity-${metadata.name}`,
+            editionId: interval.start.limit,
+          },
+          entityTypeId: metadata.schema as VersionedUrl,
+          temporalVersioning: {
+            transactionTime: interval,
+            decisionTime: interval,
+          },
+        },
       };
 
-      // reset data source input when switching blocks
-      if (example) {
-        if (typeof example.entityId === "string") {
-          setExampleEntityId(example.entityId);
-          if (example.properties && typeof example.properties === "object") {
-            setText(JSON.stringify(example.properties, undefined, 2));
-          }
+      let entityProperties: EntityPropertiesObject | undefined;
+
+      // Prefer the `variants` field if it exists
+      const selectedVariant = metadata.variants?.[blockVariantsTab];
+      if (selectedVariant) {
+        if (selectedVariant.examples) {
+          [entityProperties] = selectedVariant.examples;
         } else {
-          setText(JSON.stringify(example, undefined, 2));
-          setExampleEntityId(`test-entity-${metadata.name}`);
+          entityProperties = selectedVariant.properties;
         }
       } else {
-        setText("{}");
+        const examples = metadata.examples;
+        if (examples) {
+          [entityProperties] = examples;
+        }
+      }
+
+      if (entityProperties) {
+        exampleEntity.properties = entityProperties;
+        setEntity(exampleEntity);
+        return;
+      }
+    }
+
+    if (exampleGraph) {
+      const exampleEntity = exampleGraph.entities.find(
+        (graphEntity) =>
+          graphEntity.metadata.recordId.entityId ===
+            exampleGraph.blockEntityRecordId.entityId &&
+          graphEntity.metadata.recordId.editionId ===
+            exampleGraph.blockEntityRecordId.editionId,
+      );
+
+      if (exampleEntity) {
+        setEntity(exampleEntity);
+      } else {
+        throw new Error(
+          `Unable to find an entity for block ${metadata.name} in the example graph`,
+        );
       }
+    } else if (!metadata.variants && !metadata.examples) {
+      // The block hasn't supplied an example graph or a schema, so as long as there aren't any variants or examples
+      // we can pass it anything as the type should be unconstrained. We can't pass in a variant or an example as we
+      // don't know what the type should be.
```

Comment:
This captures one of the strange edge-cases we're being forced to handle due to `variants` and `examples` just having `properties`. (It also means we can't create link entities)

cc: @CiaranMn